flopy/utils/util_array.py

```diff
@@ -2808,28 +2808,17 @@ def array2string(shape, data, fortran_format="(FREE)", python_format=None):
                     + "   [column_length, fmt]\n"
                     + "    e.g., [10, {0:10.2e}]"
                 )
-        if ncol % column_length == 0:
-            linereturnflag = False
-        else:
-            linereturnflag = True
         # write the array to a string
-        s = ""
-        for i in range(nrow):
-            icol = 0
-            for j in range(ncol):
-                try:
-                    s = s + output_fmt.format(data[i, j])
-                except Exception as e:
-                    raise Exception(
-                        "error writing array value"
-                        + "{0} at r,c [{1},{2}]\n{3}".format(
-                            data[i, j], i, j, str(e)
-                        )
-                    )
-                if (j + 1) % column_length == 0.0 and (j != 0 or ncol == 1):
-                    s += "\n"
-            if linereturnflag:
-                s += "\n"
+        len_data = len(data.flatten())
```

Comment:
avoid making a copy of an array
```suggestion
        len_data = data.size
```